### ISSUE TITLE
It seems that the batchsize matters much for beit?

### ISSUE THREAD

**Issue by Zhaoyi-Yan** [microsoft/unilm#676]
I am trying to reproduce the results of Beit of base.
![图片](https://user-images.githubusercontent.com/29570378/161747550-8c21c048-cd07-4850-8ca3-5b5d18bb2871.png)

I used 4 DGX-2 boxes for `22k` pretraining and intermediate finetuning. And I do not change the batchsize.
Then I used 2 DGX-2 boxes for finetuing on 1k with batchsize **128**.
I obtained `83.4; 83.5` top1 acc for `beit_base_patch16_224_pt22k` and `beit_base_patch16_224_pt22k_ft22k`.
Later, I change batchsize to be **32** with only one DGX-2 box for `beit_base_patch16_224_pt22k_ft22k` training, it delivers `84.6`.

For `384*384` of `beit_base_patch16_224_pt22k_ft22k`, the performance is `86.2`.

I have several questions:
1. Do I need to keep the same total batchsize no matter how many gpus I used ?
2. I disabled the `deepspeed` to avoid errors, does that caused the performance degradation ?
3. Does the total batchsize matters a lot when doing pretraining ? Because the pretraining on 22k is very time-comsuming and I need to use 4 or 6 DGX-2 boxes to make the training  time bearable for me.
4. Could you give some advices for pretraining/finetuning of the changes on super-parameters?

Many thanks in advance!

**Comment by addf400** [comment]
Hi @Zhaoyi-Yan ,
  Q1: I recommend ensuring that the total batch size is kept the same and that the same learning rate is used to ensure reproducibility.
  Q2: Deepspeed can speech up the training by using APEX O2 instead of the APEX O1 in torch.amp, more details can be found at this [URL](https://nvidia.github.io/apex/amp.html#opt-levels).
  Q3 & Q4: We didn't try to adjust the learning rate in pretraining because it is costly and time-consuming you mentioned. It is relatively easy and fast to adjust the learning rate in the finetuning stage. We follow BERT to try several different learning rates in fine-tuning stage, as shown in Appendix H of [our paper](https://openreview.net/pdf?id=p-BhZSz59o4). It will need to reduce the learning rate and increase the layer decay when fine-tuning the models intermediate fine-tuned on ImageNet-22k. The more fine-tuning scripts can be found at this [URL](https://github.com/microsoft/unilm/blob/master/beit/get_started_for_image_classification.md#fine-tuning).
